lstchain/scripts/lstchain_dl2_to_dl3.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+"""
+Script to create DL3 of single (merged) DL2 run along with IRFs generated from lstchain_mc_dl2_to_irf.py.
+
+The selection cuts applied are the same as those used in generating IRFs.
+
+TODO:
+Adding filter for size of files after cuts, to have significant number of events
+Generalize source_fov_offset binning
+
+ - Input: Path where the merged DL2 data HDF5 file is present
+          Source name
+          IRFs
+ - Output: DL3 of the input DL2 data file in fits format.
+
+Usage:
+$> python lstchain_dl2_to_dl3.py
+--input-data ./DL2/dl2_LST-1.Run*.h5
+--output-fits-dir ./DL3/
+--add-irf True
+--input-irf-file ./IRF/irf.fits.gz
+--source-name Crab
+--obs-mode WOBBLE
+--config ../../data/data_selection_cuts.json
+"""
+
+import os
+import json
+from distutils.util import strtobool
+import numpy as np
+import argparse
+from pathlib import Path
+import logging
+import sys
+
+from lstchain.irf import create_event_list
+from lstchain.io import read_data_dl2_to_QTable, read_configuration_file, get_standard_config
+from lstchain.reco.utils import filter_events
+from lstchain.paths import run_info_from_filename
+
+from astropy.io import fits
+import astropy.units as u
+from astropy.coordinates.angle_utilities import angular_separation
+
+from pyirf.utils import calculate_source_fov_offset, calculate_theta
+
+log = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser(description="DL2 to DL3")
+
+# Required arguments
+parser.add_argument('--input-data', '-d', type=Path,
+                    dest='input_data',
+                    help='path to merged DL2 data HDF5 file',
+                    default=None, required=True
+                    )
+
+parser.add_argument('--output-fits-dir', '-o', type=Path,
+                    dest='output_fits_dir',
+                    help='path to output fits files',
+                    default=None, required=True
+                    )
+
+parser.add_argument('--add-irf', '-add-irf', action='store',
+		            type=lambda x: bool(strtobool(x)), dest='add_irf',
+                    help='Boolean: True to add IRF to DL3',
+                    default=False, required=True
+                    )
+
+parser.add_argument('--input-irf-file', '-irf', type=Path, dest='irf',
+                    help='Path to the fits.gz file of IRFs',
+                    default=None, required=False
+                    )
+
+parser.add_argument('--source-name', '-s', type=str,
+                    dest='source_name',
+                    help='Name of the source',
+                    default='Crab', required=True
```

Comment:
Default and required are mutually exclusive. What is it?

Reply:
Ah yes! I will make default None, to have this argument required.

Reply:
No, you just don't need a default if `required=True`.

Reply:
Or, I can remove the default altogether for this and the obs_mode argument. WIll that be better?

Reply:
Yes. In python you dont need to give arguments when they have a default.